apis/field_error.go

```diff
@@ -108,3 +108,12 @@ func ErrInvalidValue(value string, fieldPath string) *FieldError {
 		Paths:   []string{fieldPath},
 	}
 }
+
+// ErrMutuallyExclusiveFields is a variadic helper method for constructing a FieldError for a set of mutually exclusive fields.
+// Must have at least one set, but only one.
+func ErrMutuallyExclusiveFields(fieldPaths ...string) *FieldError {
+	return &FieldError{
+		Message: "mutually exclusive fields, must set only one of",
```

Comment:
There are actually multiple failure modes for this, we need this here: https://github.com/knative/serving/blob/5812169d86432d4895425a00966809223abc379c/pkg/apis/serving/v1alpha1/service_validation.go#L35-L49

Do you want to expose both?

Reply:
Perhaps `ErrMissingOneOf`, or `ErrMultipleOneOf`?

Former takes all options, latter takes just those specified.